examples/pgvector-chunk-embedder/20_clean_and_embed_json_data.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2021-2024 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import json
+import logging
+import pathlib
+import re
+
+from config import CHUNK_OVERLAP
+from config import CHUNK_SIZE
+from config import DOCUMENTS_JSON_FILE_LOCATION
+from config import EMBEDDINGS_PKL_FILE_LOCATION
+from nltk.tokenize import word_tokenize
+from sentence_transformers import SentenceTransformer
+from vdk.api.job_input import IJobInput
+
+log = logging.getLogger(__name__)
+
+
+def clean_text(text):
+    """
+    Prepares text for NLP tasks (embedding and RAG) by standardizing its form. This involves
+    converting to lowercase (uniformity) and removing punctuation and special characters.
+
+    :param text: A string containing the text to be processed.
+    :return: The processed text as a string.
+    """
+    text = text.lower()
+    # remove punctuation and special characters
+    text = re.sub(r"[^\w\s]", "", text)
+    return text
+
+
+def load_and_clean_documents(json_file_path):
+    """
+    Loads the documents and cleans them.
+
+    :param json_file_path: Path to the JSON file containing the documents.
+    :return: List of cleaned documents.
+    """
+    cleaned_documents = []
+    with open(json_file_path, encoding="utf-8") as file:
+        documents = json.load(file)
+
+    for doc in documents:
+        if "data" in doc:
+            cleaned_text = clean_text(doc["data"])
+            cleaned_documents.append([cleaned_text])
+
+    print(len(cleaned_documents))
+    return cleaned_documents
+
+
+def split_documents_into_chunks(
```

Comment:
I think it would be better to make it a separate job (or step at least). 
You can generate new data file  with data chunked as output. 
This file can be picked off pgvector-embedder

so the pipeline can be 
[confluence-reader] -> [chunker] -> [pgvector-embedder] 

My current understanding is that there is a lot of customization in how data is chunked.
[This article](https://medium.com/@anuragmishra_27746/five-levels-of-chunking-strategies-in-rag-notes-from-gregs-video-7b735895694d) is some outline of some strategies 
And it is more often than not somewhat coupled with the source. For example confluence type of chunking might be utiliza strategy of chunking based on headers (sections) in the page. 

we can have some default chunking strategy that is source agnostic - which is what you are building here as it make sense to have one to get started.

Reply:
I've changed the structure of the job by extracting the cleaning, chunking and embedding as separate job steps. 
Added a property accounting for the chunking strategy that is easily extensible - by default it is the current fixed size chunking.